torch/distributed/_tensor/debug/comm_mode.py

```diff
@@ -59,31 +59,39 @@ class ModuleParamaterShardingTracker(ModuleTracker):
 
     def __init__(self):
         super().__init__()
+        self.module_depth_dict = {}
         self.module_parameters_dict = {}
         self.sharding_dict = {}
+        self.name = ""
 
     def _fw_pre_hook(self, mod, input):
-        name = super()._get_mod_name(mod)
-        super()._get_append_fn(name, False)()
+        self.name = super()._get_mod_name(mod)
+
+        # contains information about module ordering and depth in the module tree
+        self.module_depth_dict[self.name] = len(self.parents)
+
+        super()._get_append_fn(self.name, False)()
```

Comment:
curious: could you explain what this `_get_append_fn` does? it could be difficult to understand for users who're not familiar with `ModuleTracker`.